### ISSUE TITLE
While trying for prediction iit looking for  SAM model rather than yoloV8

### ISSUE THREAD

**Issue by chanakya009** [ultralytics/ultralytics#6245]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

!yolo predict model='/Sample Files/yolov8m.pt' source='/Images/LAY-OUT.png'  i am actually using this command to predict my new image on my trained data which has given me yolov8m.pt but it is looking for SAM Model try to fix this i am having yolov8 in my system i have check everything all the files are there but still its over looking for SAM Model insted of yolov8

### Environment

Ultralytics YOLOv8.0.207 🚀 Python-3.11.5 torch-2.1.0 CPU (Apple M3 Pro)
Setup complete ✅ (11 CPUs, 18.0 GB RAM, 90.9/460.4 GB disk)

### Minimal Reproducible Example

from ultralytics import YOLO

# Load a model
model = YOLO('yolov8m.pt')  # load a pretrained model (recommended for training)

# Train the model with 2 GPUs
results = model.train(data='/Sample Files/YoloV8m.yaml', epochs=300, imgsz=640, device='mps')
after running this i on my custome data i have got yolov8m.pt files into file system and after that i am using this yolov8m.pt file and using the best.pt from my trained folder i am using this command for my prediction !yolo predict model='/Sample Files/yolov8m.pt' source='/Images/LAY-OUT.png'. now it is looking for SAM model other than yolov8 model tough i am not having SAM model in my system but i found it part of tensorflow

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@chanakya009 oh this is interesting, thanks for reporting this. I've tried to make model loading smart by parsing the model path, but you seem to have uncovered an edge case here that I didn't think of. I will review my code here and hopefully push a fix for this soon.

**Comment by glenn-jocher** [comment]
@chanakya009 great news 😃! Your original issue may now be resolved ✅ in PR #6257. 

Loading alternate model types like SAM from the CLI can be tricky though, the rule is that for YOLO to recognize it correctly it needs to have "sam" in the filename. Before we were picking it up along the entire path, hence your bug, now it's limited to just the filename itself, and your YOLO model should load correctly.

To get this update:
- **[Git](https://github.com/ultralytics/ultralytics)** – Run `git pull` from within your `ultralytics/` directory or run `git clone https://github.com/ultralytics/ultralytics` again
- **[Pip](https://pypi.org/project/ultralytics/)** – Update with `pip install -U ultralytics`
- **[Notebooks](https://github.com/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb)** – Check out the updated notebooks  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/ultralytics)** – Run `sudo docker pull ultralytics/ultralytics:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and letting us know. Please confirm if this update fixes the issue for you, and don't hesitate to report any other **issues** you find or **feature requests** you may have. Happy training with YOLOv8 🚀!

**Comment by sergiev** [comment]
I ran into the same problem when my model path included the word `down*sam*ple`, so I think checking for a three-character substring is too weak a condition. @glenn-jocher, what do you think of this solution:
```python
    import re
    def is_model_type(path: str, model_type: str) -> bool:
        regex = re.compile(rf'[^a-z]{model_type.lower()}[^a-z]')
        return bool(regex.search(path.lower()))
    overrides['model'] = model
    if is_model_type(model, 'rtdetr'):  # guess architecture
        from ultralytics import RTDETR
        model = RTDETR(model)  # no task argument
    elif is_model_type(model, 'fastsam'):
        from ultralytics import FastSAM
        model = FastSAM(model)
    elif is_model_type(model, 'sam'):
        from ultralytics import SAM
        model = SAM(model)
    else:
        from ultralytics import YOLO
        model = YOLO(model, task=task)
```

Here we treat all substrings of (consecutive letters) as words, and if you need to specify the usage of SAM, you can specify it with any non-alphabetical delimiters you want.

Current logic blocks misinterprets all words containing the substring `sam`, e.g. I have to use weird splitting tactics like `downsa_mple`.

**Comment by glenn-jocher** [comment]
Thanks for the code! In my fix PR I've updated to scan only stem for  the string occurrence rather than the full file path. Would this resolve your case?

**Comment by glenn-jocher** [comment]
@sergiev you're welcome! If you encounter any other issues or have further questions about YOLOv8, feel free to reach out. Happy detecting!

**Comment by chanakya009** [comment]
> @chanakya009 oh this is interesting, thanks for reporting this. I've tried to make model loading smart by parsing the model path, but you seem to have uncovered an edge case here that I didn't think of. I will review my code here and hopefully push a fix for this soon.

I have checked its working fine now thanks for quick updates and will again reach you guys if i find any in my YOLO implemanation

**Comment by chanakya009** [comment]
> 👋 Hello @chanakya009, thank you for your interest in YOLOv8 🚀! We recommend a visit to the [YOLOv8 Docs](https://docs.ultralytics.com) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.
> 
> If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug it.
> 
> If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/yolov5/tutorials/tips_for_best_training_results/).
> 
> Join the vibrant [Ultralytics Discord](https://ultralytics.com/discord) 🎧 community for real-time conversations and collaborations. This platform offers a perfect space to inquire, showcase your work, and connect with fellow Ultralytics users.
> ## Install
> 
> Pip install the `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/requirements.txt) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/).
> 
> ```shell
> pip install ultralytics
> ```
> 
> ## Environments
> 
> YOLOv8 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
> 
>     * **Notebooks** with free GPU: [![Run on Gradient](https://camo.githubusercontent.com/c9931a1689c37ab786edd3e1e5f59b9a6f7d097628c4689ce2432563ef884524/68747470733a2f2f6173736574732e706170657273706163652e696f2f696d672f6772616469656e742d62616467652e737667)](https://console.paperspace.com/github/ultralytics/ultralytics) [![Open In Colab](https://camo.githubusercontent.com/84f0493939e0c4de4e6dbe113251b4bfb5353e57134ffd9fcab6b8714514d4d1/68747470733a2f2f636f6c61622e72657365617263682e676f6f676c652e636f6d2f6173736574732f636f6c61622d62616467652e737667)](https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb) [![Open In Kaggle](https://camo.githubusercontent.com/a08ca511178e691ace596a95d334f73cf4ce06e83a5c4a5169b8bb68cac27bef/68747470733a2f2f6b6167676c652e636f6d2f7374617469632f696d616765732f6f70656e2d696e2d6b6167676c652e737667)](https://www.kaggle.com/ultralytics/yolov8)
> 
>     * **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
> 
>     * **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
> 
>     * **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) [![Docker Pulls](https://camo.githubusercontent.com/7401823bed1473d6db1ff60b933d51cd1534c448488f06c446c322227d31d966/68747470733a2f2f696d672e736869656c64732e696f2f646f636b65722f70756c6c732f756c7472616c79746963732f756c7472616c79746963733f6c6f676f3d646f636b6572)](https://hub.docker.com/r/ultralytics/ultralytics)
> 
> 
> ## Status
> 
> [![Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg)](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule)
> 
> If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLOv8 [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

U Guys are doing grate work and updates pretty quick for me we are working on research project where we are buliding the  OCR project using Yolov8 of Ultralytics and as per our research ours is the 1st ever project in complete yolo and OCR on Civil Construction and happy to tell with your quick updates we have completed basic prototype for few of the class Objects where we have done annotation on CVAT once again thanks for fix the issue i have raised @glenn-jocher

**Comment by glenn-jocher** [comment]
Hello @chanakya009, I'm thrilled to hear that the update has helped and that your OCR project on civil construction is making great progress with YOLOv8! 🚀 It's fantastic to learn about innovative applications like yours. Thank you for choosing Ultralytics and for your kind words. Remember, we're here to support you, so reach out anytime you need assistance. Good luck with your research project, and keep up the exceptional work! 🏗️📑

**Comment by chanakya009** [comment]
Hi Glenn,

Our project went great and we have got exponential prediction rate by using
your V8 model but one thing i want to tell you as ours is small custom
dataset it was fine running 100-300 epoch in M3 silicon chip and i want to
tell you guys one thing that it will be much easier that you introduce
Base64 image encryption concept into your model where the encrypted image
will and its corresponding label files are stored in Graph Database which
makes your model will run more epochs in less time and i can see feature
object detection will be done very simpler manner using graph database how
it is using for LLM in language model.

I alway will be there for you guys for testing and providing updates on how
you model is working if you implement this feature

I hope this will help you guys,

Thanks,
Chanakya.K

**Comment by glenn-jocher** [comment]
@chanakya009 hello! It seems like there might be a misunderstanding with the model file you're trying to use for prediction. The `yolov8m.pt` file should be the correct model if it was the output of your training process. The mention of a SAM model could be due to a misconfiguration or an unexpected error.

To resolve this, please ensure that the `yolov8m.pt` file is indeed the trained model file and that it's located in the correct directory. If the issue persists, it would be helpful to have the exact error message or any additional logs that are output when you attempt to run the prediction command.

If you continue to experience issues, please open a new issue on the YOLOv8 GitHub repository with the full error message and any relevant details so we can assist you further. 🛠️
